### PR TITLE
Des trucs, faut squash maintenant :)

### PR DESCRIPTION
**Pull request by destiny06**

### CODE REVIEW

File: src/client/lobby/BasicLobbyClient.cpp [hunk 1]
```diff
@@ -9,8 +9,9 @@ BasicLobbyClient::BasicLobbyClient(SocketClient& s)
       myself(NULL),
       externalListener(NULL) {
 
-    /* TODO: Workaround waiting for JOIN_GAME implementation */
-    clients.push_back(Client(0,"Nobody"));
+    /* TODO: Workaround */
+    myself = new Client(0,"");
+    clients.push_back(myself);
 }
```
Comment (Telsanyr) [inline src/client/lobby/BasicLobbyClient.cpp:15]: Does it prevent a copy ? Do we need to take care of the pointer or is it managed by the list when removed ?

File: src/client/lobby/BasicLobbyClient.cpp [hunk 2]
```diff
@@ -21,16 +22,22 @@ void BasicLobbyClient::setExternalListener(LobbyClientListener* listener){
...
+             * thus we won't know all the potential clients until the game start*/
+
+            /* TODO check that our join worked (similar to add name technic ?)*/
+
+            Client *client = retrieveClientFromName(action.name());
+
+            if (client == nullptr) {
+                client = new Client(0, action.name());
+                clients.push_back(client); /* TODO: Implement ID */
             }
```
Comment (Telsanyr) [inline src/client/lobby/BasicLobbyClient.cpp:41]: Should I refactor every NULL into nullptr for consistency ?

File: src/client/lobby/BasicLobbyClient.cpp [hunk 3]
```diff
@@ -102,12 +111,17 @@ void BasicLobbyClient::onServerAction(const PServerAction& action){
...
-    for(std::list<Client>::iterator it = clients.begin(); it != clients.end(); it++){
-        if(it->name == name){
-            return &(*it);
-        }
+
+    std::list<Client*>::iterator it = std::find_if(
+    clients.begin(),
+    clients.end(),
+            [name](const Client* c) {return c->name == name; });
+
```
Comment (Telsanyr) [inline src/client/lobby/BasicLobbyClient.cpp:119]: Why is it better to use find_if with lambda fct here ? Is it more "standard" ?

File: src/client/lobby/BasicLobbyClient.cpp [hunk 4]
```diff
@@ -102,12 +111,17 @@ void BasicLobbyClient::onServerAction(const PServerAction& action){
...
+    if (it != clients.end()) {
+        return *it;
+    } else {
+        return NULL;
     }
```
Comment (Telsanyr) [inline src/client/lobby/BasicLobbyClient.cpp:124]: nullptr ?

File: src/client/network/WebSocketClient.cpp [hunk 5]
```diff
@@ -30,17 +30,15 @@ void WebSocketClient::disconnect(){
...
 
-
     std::cout << "[WebSocket-Logger]: Sending message: [";
-    for(int i=0; i<s.size(); i++){
+    for(int i=0; i<s.size()-1; i++){
         std::cout << (int) s[i] << "-";
     }
-    std::cout << "], size( "<< s.size() << ")\n";
-
+    std::cout << (int) s[s.size()-1] << "], size( "<< s.size() << ")\n";
```
Comment (Telsanyr) [inline src/client/network/WebSocketClient.cpp:42]: Good job my friend !

**Comment by Telsanyr** [comment]
Good job, ty for const args & comment cleaning. I just need you to answer my C++ noob questions !

File: src/client/lobby/BasicLobbyClient.cpp [hunk 6]
```diff
@@ -21,16 +22,22 @@ void BasicLobbyClient::setExternalListener(LobbyClientListener* listener){
...
-                break;
+            /* TODO Basically we can only know players that are joining after we were connected
+             * thus we won't know all the potential clients until the game start*/
+
+            /* TODO check that our join worked (similar to add name technic ?)*/
+
+            Client *client = retrieveClientFromName(action.name());
+
+            if (client == nullptr) {
+                client = new Client(0, action.name());
```
Comment (destiny06) [inline src/client/lobby/BasicLobbyClient.cpp:39]: coucou

Comment (Telsanyr) [reply src/client/lobby/BasicLobbyClient.cpp:39]: Hello my friend

File: src/client/network/WebSocketClient.cpp (hunk 5 above)
Comment (destiny06) [reply src/client/network/WebSocketClient.cpp:42]: thanks my friend

File: src/client/lobby/BasicLobbyClient.cpp (hunk 4 above)
Comment (destiny06) [reply src/client/lobby/BasicLobbyClient.cpp:124]: ouais bon...

File: src/client/lobby/BasicLobbyClient.cpp (hunk 2 above)
Comment (destiny06) [reply src/client/lobby/BasicLobbyClient.cpp:41]: ouaip we should

File: src/client/lobby/BasicLobbyClient.cpp (hunk 1 above)
Comment (destiny06) [reply src/client/lobby/BasicLobbyClient.cpp:15]: "normally" managed by the list when removed, but this is temporary, we need to reshape a bit the way we handl "myself"

File: src/client/lobby/BasicLobbyClient.cpp (hunk 3 above)
Comment (destiny06) [reply src/client/lobby/BasicLobbyClient.cpp:119]: C++11 is providing to you features my friend, let's use them my friend.
Moreover I spent 1 hour to be able to do my first lambda the first time, I need to capitalise on it

**Pull request merged**